lib/capybara/cuprite/browser/web_socket.rb

```diff
@@ -51,9 +46,24 @@ def on_message(event)
         @logger&.puts("    <<< #{event.data}\n")
       end
 
+      def on_close(_event)
+        @messages.close
+        @thread.kill
+      end
+
+      def send(data)
+        json = data.to_json
+        @driver.text(json)
+        @logger&.puts("\n\n>>> #{json}")
+      end
+
       def write(data)
         @sock.write(data)
       end
+
+      def close
+        @driver.close
```

Comment:
Should Socket#close also call `@thread.join`, just like Client#close above? When I was reading the examples for websocket-driver I noticed that they do that in their socket class but curprite does not

Reply:
`websocket-driver` has two versions of `close`. One is event-driven https://github.com/faye/websocket-driver-ruby/blob/master/examples/tcp_client.rb#L25 and another one is plain close like the one we use https://github.com/faye/websocket-driver-ruby/blob/master/examples/tcp_client.rb#L42. I think it's pretty much close to what they have in example. Though `@thread.join` doesn't help as thread can be blocked on IO and they have to use `kill` as we do.